numpy/lib/tests/test_format.py

```diff
@@ -728,13 +728,13 @@ def test_read_magic():
 def test_read_magic_bad_magic():
     for magic in malformed_magic:
         f = BytesIO(magic)
-        yield raises(ValueError)(format.read_magic), f
+        raises(ValueError)(format.read_magic)(f)
 
 
 def test_read_version_1_0_bad_magic():
     for magic in bad_version_magic + malformed_magic:
         f = BytesIO(magic)
-        yield raises(ValueError)(format.read_array), f
+        raises(ValueError)(format.read_array)(f)
```

Comment:
No value in using the `raises` decorator here now that it's not a yield test - just use `assert_raises`